frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts

```diff
@@ -162,6 +167,59 @@ const osUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
   if (!isWindows && windowsTools) {
     dispatch(vmWizardInternalActions[InternalActionType.RemoveStorage](id, windowsTools.id));
   }
+
+  const iUserTemplates = iGetLoadedCommonData(state, id, VMWizardProps.userTemplates);
+  const iCommonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+  const template = toShallowJS(iGetRelevantTemplate(iUserTemplates, iCommonTemplates, selectors));
+  const pvcName = getOperatingSystemDataVolumeAnnotation(template, selectors?.os);
+
+  dispatch(
+    vmWizardInternalActions[InternalActionType.UpdateVmSettingsField](
+      id,
+      VMSettingsField.CLONE_BASE_DISK_IMAGE,
+      {
+        value: !!pvcName,
+      },
+    ),
+  );
+};
+
+const mountBaseImageUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
+  const state = getState();
+  if (!hasVMSettingsValueChanged(prevState, state, id, VMSettingsField.CLONE_BASE_DISK_IMAGE)) {
+    return;
+  }
+  const userTemplateName = iGetVmSettingValue(state, id, VMSettingsField.USER_TEMPLATE);
+  if (userTemplateName) {
+    return;
+  }
+
+  const mountBaseImage = iGetVmSettingValue(state, id, VMSettingsField.CLONE_BASE_DISK_IMAGE);
+  if (mountBaseImage) {
+    dispatch(
+      vmWizardInternalActions[InternalActionType.UpdateVmSettings](id, {
+        [VMSettingsField.PROVISION_SOURCE_TYPE]: {
+          isHidden: asHidden(true, VMSettingsField.CLONE_BASE_DISK_IMAGE),
+          value: ProvisionSource.DISK.toString(),
+        },
+        [VMSettingsField.CONTAINER_IMAGE]: {
+          isHidden: asHidden(true, VMSettingsField.PROVISION_SOURCE_TYPE),
+        },
+        [VMSettingsField.IMAGE_URL]: {
+          isHidden: asHidden(true, VMSettingsField.PROVISION_SOURCE_TYPE),
+        },
+      }),
+    );
+  } else {
+    dispatch(
```

Comment:
it is better to write this as a single dispatch so we are sure all fields are set to desired state and the values depend on the right prerequisites/values

Reply:
done